### PR TITLE
HLM-6019 download template

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Hypothesis.js [hunk 1]
```diff
@@ -520,7 +520,8 @@ const Input = React.memo(({ item, setAssumptions, t, disabled = false }) => {
 
   const inputChangeHandler = useCallback(
     (e) => {
-      if ((e.target.value <= 0 || e.target.value > 10000000000) && e.target.value !== "" && ["+", "e"].includes(e.target.value)) return;
+      if( e.target.value.includes("+") || e.target.value.includes("e") ) return
+      if ((e.target.value <= 0 || e.target.value > 10000000000) && e.target.value !== "") return;
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Hypothesis.js:524]: this is not working

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/Module.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { ProviderContext } from "./utils/context";
 
 const MicroplanningModule = ({ stateCode, userType, tenants }) => {
   
-  const moduleCode = ["Microplanning"];
+  const moduleCode = ["Microplanning", "campaignmanager"];
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/Module.js:15]: yes it is

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js [hunk 3]
```diff
@@ -51,12 +51,12 @@ export const JsonPreviewInExcelForm = (props) => {
...
                         key={cellIndex}
                         style={{
-                          ...(!isNaN(rowData[cellIndex]) && isFinite(rowData[cellIndex]) ? { textAlign: "end" } : {}),
+                          ...(rowData[cellIndex] || rowData[cellIndex] === 0 ? !isNaN(rowData[cellIndex]) && isFinite(rowData[cellIndex]) ? { textAlign: "end" } :{}: {}),
                           ...( error? {backgroundColor:"rgb(250,148,148)"} : {}),
                         }}
                         title={error? t(error):undefined}
                       >
-                        {rowData[cellIndex] || ""}
+                        {rowData[cellIndex] || rowData[cellIndex] === 0 ?rowData[cellIndex] : ""}
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/JsonPreviewInExcelForm.js:59]: Number.isNaN give 21fasf as numaric

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/createTemplate.js [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+  // Add the hierarchy as the first row of the sheet
+  sortedBoundaryDataForXlsxSheet = [[...hierarchy], ...sortedBoundaryDataForXlsxSheet];
+
+  // Determine the maximum row length to ensure all rows have the same length
+  const topIndex = Math.max(...sortedBoundaryDataForXlsxSheet.map((row) => row.length)) - 1;
+
+  // Ensure all rows are of the same length by filling them with empty strings
+  sortedBoundaryDataForXlsxSheet = sortedBoundaryDataForXlsxSheet.map((item) => {
+    if (!item) {
+      item = [];
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/createTemplate.js:99]: Doesnt fit the usecase

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/createTemplate.js [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+  // Initialize the array to hold new data
+  let newXlsxData = [];
+
+  // Recursive function to convert boundary data into sheet format
+  const convertBoundaryDataToSheets = (boundaryData, currentBoundaryPredecessor = [], hierarchyAccumulator = [], dataAccumulator = []) => {
+    // Return if boundary data is not valid or not an array
+    if (!boundaryData || !Array.isArray(boundaryData)) return;
+
+    // Clone the current boundary predecessor to avoid modifying the original data
+    let rowData = _.cloneDeep(currentBoundaryPredecessor);
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/createTemplate.js:56]: but these are getting reassigned later

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**